src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1682,13 +1584,42 @@ public unsafe void HandleEvents(Interop.Sys.SocketEvents events)
 
             if ((events & Interop.Sys.SocketEvents.Read) != 0)
             {
-                _receiveQueue.Complete(this);
+                _receiveQueue.HandleEvent(this);
             }
 
             if ((events & Interop.Sys.SocketEvents.Write) != 0)
             {
-                _sendQueue.Complete(this);
+                _sendQueue.HandleEvent(this);
             }
         }
+
+        //
+        // Tracing stuff
+        //
+
+        // To enabled tracing:
+        // (1) Add reference to System.Console in the csproj
+        // (2) #define SOCKETASYNCCONTEXT_TRACE
+
+#if SOCKETASYNCCONTEXT_TRACE
+        public const bool TraceEnabled = true;
+#else
+        public const bool TraceEnabled = false;
```

Comment:
instead of the bool can you just adorn `Trace()` with `[System.Diagnostics.Conditional("SOCKETASYNCCONTEXT_TRACE")]  ` or suchlike? 
Then callers don't need to check the flag, they just call Trace. If the code was not compiled with this define, both the method and the callers are compiled away. (You could also make the Console reference conditional on the define)

This assumes that you don't need to change the flag without recompiling.